### PR TITLE
fix: issue where deploy failures would not enrichment [APE-1142]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/test_compiler.py [hunk 1]
```diff
@@ -400,13 +402,24 @@ def test_enrich_error_when_custom(compiler, project, owner, not_owner, connectio
...
     # Deploy so Ape know about contract type.
-    contract = owner.deploy(project.HasError)
+    contract = owner.deploy(project.HasError, 1)
     with pytest.raises(contract.Unauthorized) as err:
         contract.withdraw(sender=not_owner)
 
     assert err.value.inputs == {"addr": not_owner.address, "counter": 123}
 
 
+@pytest.mark.skipif(APE_VERSION <= Version("0.6.11"), reason="Fails without fixes on main")
```
Comment (fubuloubu) [inline tests/test_compiler.py:412]: Is this still necessary?

Comment (antazoey) [reply tests/test_compiler.py:412]: it is until we release 0.6.12

File: tests/test_compiler.py [hunk 2]
```diff
@@ -400,13 +402,24 @@ def test_enrich_error_when_custom(compiler, project, owner, not_owner, connectio
...
+def test_enrich_error_when_custom_in_constructor(compiler, project, owner, not_owner, connection):
+    contract_type = compiler.compile((project.contracts_folder / "HasError.sol",))[0]
+
+    # Deploy so Ape know about contract type.
+    with pytest.raises(contract_type.Unauthorized) as err:
```
Comment (dtdang) [inline tests/test_compiler.py:417]: The test seems to be failing for me locally `AttributeError: 'ContractType' object has no attribute 'Unauthorized'`

File: tests/test_compiler.py (hunk 1 above)
Comment (antazoey) [reply tests/test_compiler.py:413]: ok just made it require a release

File: tests/test_compiler.py (hunk 2 above)
Comment (antazoey) [reply tests/test_compiler.py]: ok fixed now

**Review by dtdang — APPROVED** [review]

**Pull request merged**